### ISSUE TITLE
Constant Value Checker: StringVal incorrect handling of null values

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#1169]
When myInteger is null,
```java
  "myInteger=" + myInteger
```
may evaluate to
```java
  "myInteger=null"
```
Also, when myInteger is declared as
```java
  @StringVal(22) String myInteger;
```
then
```java
  "myInteger=" + myInteger
```
may evaluate to two values:
```java
  "myInteger=22"
  "myInteger=null"
```
The Value Checker does not handle this soundly.

The below test case illustrates this problem.

It is necessary for the programmer to specify whether null is possible, and to change the Constant Value Checker to take account of those specifications.

Here are two ways to solve the problem.  Other solutions are welcome.

 1. Have the Value Checker read and trust Nullness Checker annotations if they are present.  At a conversion to String, the nullness annotations are looked up and utilized, adding "null" to the set as needed.  (The Value Checker needs to trust the Nullness Checker because we don't want to make a checker in the `framework` package depend on a checker in the `checker` package.  Careful users would want to run the Nullness Checker to verify the nullness annotations.  Maybe the Constant Value Checker would issue a warning if it depended on nullness annotations and the Nullness Checker was not currently being run.)

 2. Change the @StringVal annotation's documentation so that users are required to include "null" in its argument.  The Constant Value Checker must perform a subset of the work done by the Nullness Checker, so that it can create appropriate StringVal annotations.  Unfortunately, Java doesn't permit null to appear in the argument along with Literal strings, so the annotation needs to use "null" instead.  Furthermore, this proposal doesn't enable @IntVal to specify when an Integer might be null.  So, this proposal seems bad.

Other suggestions?

```java
// Type-checking this program with the Nullness and Constant Value checkers
// yields no errors or warnings:
//   $ch/bin/javac -g StringValNull.java -processor index,nullness
// but should yield the two marked return.type.incompatible errors.

import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.common.value.qual.IntVal;
import org.checkerframework.common.value.qual.StringVal;

public class StringValNull {

  public static void main(String[] args) {
    @StringVal("itsValue") String nbleString = null;
    @StringVal("itsValue") String nnString = "itsValue";

    System.out.println(toString1(nbleString));
    System.out.println(toString2(nbleString));

    System.out.println(toString1(nnString));
    System.out.println(toString2(nnString));
    System.out.println(toString3(nnString));

    @IntVal(22) Integer nbleInteger = null;
    @IntVal(22) Integer nnInteger = 22;

    System.out.println(toString4(nbleInteger));
    System.out.println(toString5(nbleInteger));

    System.out.println(toString4(nnInteger));
    System.out.println(toString5(nnInteger));
    System.out.println(toString6(nnInteger));
  }

  static @StringVal("arg=itsValue") String toString1(@Nullable @StringVal("itsValue") String arg) {
    //:: error: (return.type.incompatible)
    return "arg=" + arg;
  }

  static @StringVal({"arg=itsValue", "arg=null"}) String toString2(@Nullable @StringVal("itsValue") String arg) {
    return "arg=" + arg;
  }

  static @StringVal("arg=itsValue") String toString3(@StringVal("itsValue") String arg) {
    return "arg=" + arg;
  }

  static @StringVal("arg=22") String toString4(@Nullable @IntVal(22) Integer arg) {
    //:: error: (return.type.incompatible)
    return "arg=" + arg;
  }

  static @StringVal({"arg=22", "arg=null"}) String toString5(@Nullable @IntVal(22) Integer arg) {
    return "arg=" + arg;
  }

  static @StringVal("arg=22") String toString6(@IntVal(22) Integer arg) {
    return "arg=" + arg;
  }

}
```
